src/software/ai/navigator/obstacle/obstacle_test.cpp

```diff
@@ -3,231 +3,181 @@
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    Circle expected({2, 2}, 3);
+    CircleObstacle circle_obstacle(expected);
+
+    EXPECT_EQ(circle_obstacle.getCircle(), expected);
+    std::ostringstream circle_ss;
+    circle_ss << expected;
+    EXPECT_TRUE(circle_obstacle.toString().find(circle_ss.str()) != std::string::npos);
 }
 
-// obstacle with robot radius factor = 1.2, velocity projection factor = 1.4
-// centred at (-1,-2) and with (-3,2) velocity
-TEST(NavigatorObstacleTest, shifted_scaled_up_velocity_obstacle_polygon)
+TEST(NavigatorObstacleTest, rectangle_obstacle_contains)
 {
-    double TEST_EPSILON    = 1e-3;
-    Timestamp current_time = Timestamp::fromSeconds(123);
-    Robot robot    = Robot(3, Point(-1, -2), Vector(-3, 2), Angle::fromRadians(2.2),
-                        AngularVelocity::fromRadians(-0.6), current_time);
-    auto p_pointer = Obstacle::createRobotObstacleWithScalingParams(robot, 1.2, 1.4)
-                         .getBoundaryPolygon();
-
-    EXPECT_TRUE(p_pointer);
-
-    if (p_pointer)
-    {
-        Polygon pg = *p_pointer;
-        // visually verified
-        Point pt0 = Point(-1.138, -2.208);
-        Point pt1 = Point(-.889, -2.224);
-        Point pt2 = Point(-0.751, -2.016);
-        Point pt3 = Point(-0.862, -1.792);
-        Point pt4 = Point(-5.241, 1.127);
-        Point pt5 = Point(-5.518, 0.712);
-        EXPECT_DOUBLE_EQ(pg.getPoints().size(), 6);
-        EXPECT_NEAR(pg.getPoints()[0].x(), pt0.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[0].y(), pt0.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].x(), pt1.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[1].y(), pt1.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].x(), pt2.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[2].y(), pt2.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].x(), pt3.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[3].y(), pt3.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].x(), pt4.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[4].y(), pt4.y(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].x(), pt5.x(), TEST_EPSILON);
-        EXPECT_NEAR(pg.getPoints()[5].y(), pt5.y(), TEST_EPSILON);
-    }
+    Rectangle rectangle({-1, 1}, {2, -3});
+    Obstacle obstacle(std::make_shared<PolygonObstacle>(PolygonObstacle(rectangle)));
+    Point inside_point(0, -1);
+    Point outside_point(5, 5);
+
+    EXPECT_EQ(obstacle->contains(inside_point), rectangle.contains(inside_point));
+    EXPECT_EQ(obstacle->contains(outside_point), rectangle.contains(outside_point));
```

Comment:
Why not `EXPECT_TRUE(obstacle->contains(inside_point))`. Generally it seems a better choice to avoid relying on the implementation of `Rectangle` here if we can easily avoid this.

Reply:
ok fixed this and all cases